src/ensembl/src/content/app/entity-viewer/state/bookmarks/entityViewerBookmarksSlice.ts

```diff
@@ -40,6 +52,77 @@ type UpdatePreviouslyViewedPayload = {
   };
 };
 
+export const updatePreviouslyViewedEntities = createAsyncThunk(
+  'entity-viewer/updatePreviouslyViewedEntities',
+  (params: UpdatePreviouslyViewedPayload, thunkAPI) => {
+    const { genomeId, gene } = params;
+    const { getState } = thunkAPI;
+    const state = getState() as RootState;
+
+    const previouslyViewedEntities = getPreviouslyViewedEntities(
+      state,
+      genomeId
+    );
+
+    const alreadySavedGene = previouslyViewedEntities.find(
+      (entity) => entity.entity_id === gene.unversioned_stable_id
+    );
+
+    const newPreviouslyViewedEntity = {
+      entity_id: gene.unversioned_stable_id,
+      label: gene.symbol ? [gene.symbol, gene.stable_id] : gene.stable_id,
+      type: 'gene' as const
+    };
```

Comment:
Is this object necessary if `alreadySavedGene` exists?

Reply:
From my understanding thats needed when it doesn't exist, its a new gene being added. But @azangru can confirm this if i am saying the right thing here.